tree/dataframe/test/dataframe_datasetspec.cxx

```diff
@@ -832,6 +836,228 @@ TEST_P(RDatasetSpecTest, TreeInSubdir)
+      df1.DefinePerSample("lum", [](unsigned int, const ROOT::RDF::RSampleInfo &id) { return id.GetD("lum"); });
+   auto df_filtered = definepersamp.Filter("lum == 10.").Count();
+
+   auto df = RDataFrame(spec.WithGlobalRange(goodRanges[0]));
+   auto filt = df.Filter("rdfentry_ == 2");
+   auto result = filt.Take<ULong64_t>("x");
+   auto res = result.GetValue();
+   EXPECT_EQ(res[0], 2);
+
+   auto df2 = RDataFrame(spec.WithGlobalRange(goodRanges[1]));
+   auto filt2 = df2.Filter("rdfentry_ == 3");
+   auto result2 = filt2.Take<ULong64_t>("x");
+   auto res2 = result2.GetValue();
+   EXPECT_EQ(res2[0], 3);
+
+   auto df3 = RDataFrame(spec.WithGlobalRange(goodRanges[2]));
+   auto filt3 = df3.Filter("rdfentry_ == 8");
+   auto result3 = filt3.Take<ULong64_t>("x");
+   auto res3 = result3.GetValue();
+   EXPECT_EQ(res3[0], 12);
+
+   auto df4 = RDataFrame(spec.WithGlobalRange(goodRanges[3]));
+   auto filt4 = df4.Filter("rdfentry_ == 19");
+   auto result4 = filt4.Take<ULong64_t>("x");
+   auto res4 = result4.GetValue();
+   EXPECT_EQ(res4[0], 12);
+
+   EXPECT_EQ(df_final.GetValue(), 11);
+   EXPECT_EQ(df_filtered.GetValue(), 10);
+}
+
+TEST_P(RDatasetSpecTest, FromSpecRNTuple)
+{
+   const std::string prefix = "rdatasetspec_rntuple";
+   InputRNTuplesRAII file(3u, prefix);
+   auto df_fromspec = ROOT::RDF::Experimental::FromSpec("spec_rntuple.json");
+   auto df_final = df_fromspec.Filter("x > 3").Count();
+   auto definepersamp =
+      df_fromspec.DefinePerSample("lum", [](unsigned int, const ROOT::RDF::RSampleInfo &id) { return id.GetD("lum"); });
+   auto df_filtered = definepersamp.Filter("lum == 20.").Count();
+
+   EXPECT_EQ(df_final.GetValue(), 3);
+   EXPECT_EQ(df_filtered.GetValue(), 5);
+}
+
+TEST_P(RDatasetSpecTest, RNTupleWrong)
+{
+   const std::string prefix = "rdatasetspec_rntuple";
+   InputRNTuplesRAII file(1u, prefix);
+
+   auto model = ROOT::RNTupleModel::Create();
+   auto fldX = model->MakeField<int>("x");
+   auto ntpl = ROOT::RNTupleWriter::Recreate(std::move(model), "mytuple", "rntuple_wrong.root");
+   *fldX = 2;
+   ntpl->Fill();
+
+   auto samp = ROOT::RDF::Experimental::RSample("mysample", "ntuple", prefix + "0.root");
+   auto samp1 = ROOT::RDF::Experimental::RSample("mysample1", "mytuple", "rntuple_wrong.root");
+
+   RDatasetSpec spec;
+   spec.AddSample(samp);
+   spec.AddSample(samp1);
+
+   EXPECT_THROW(
+      try {
+         auto samp = ROOT::RDF::Experimental::RSample("mysample", "ntuple", prefix + "0.root");
+         auto samp1 = ROOT::RDF::Experimental::RSample("mysample1", "mytuple", "rntuple_wrong.root");
+
+         RDatasetSpec spec;
+         spec.AddSample(samp);
+         spec.AddSample(samp1);
+         auto df_error = ROOT::RDataFrame(spec);
+      }
+
+      catch (const std::runtime_error &err) {
+         EXPECT_EQ(
+            std::string(err.what()),
+            "More than one RNTuple name was found, please make sure to use RNTuples with the same RnTuple name.");
```

Comment:
Not specifically for this PR, but general for RDF devs:
I always wonder if it makes sense to really test the entire error message. In this way, we always have to touch two places if we wanted to correct small things, such as in this case:
`RnTuple --> RNTuple`

I would either just test that there is any exception (possibly of a specific type), or just test a substring such as `exc.startsWith("More than one RNTuple name");`. In this way, small adjustments or making the error message more helpful doesn't immediately fail the test.